File: src/main/java/org/zenbot/szolnok/timetable/batch/step/bus/processor/CreateBusFromTimetableItemProcessorHelper.java
package org.zenbot.szolnok.timetable.batch.step.bus.processor;

import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;
import org.zenbot.szolnok.timetable.dao.BusRepository;
import org.zenbot.szolnok.timetable.domain.Bus;
import org.zenbot.szolnok.timetable.domain.Timetable;

import java.util.ArrayList;

@Slf4j
@Component
public class CreateBusFromTimetableItemProcessorHelper {
    
    private final BusRepository busRepository;

    public CreateBusFromTimetableItemProcessorHelper(BusRepository busRepository) {
        this.busRepository = busRepository;
    }

    public Bus createBusFromTimetable(Timetable timetable) {
        log.debug("Fetching bus=[#{}] from database", timetable.getBusName());
        Bus bus = busRepository.findByBusName(timetable.getBusName());
        if (bus == null) {
            log.debug("Bus=[{}] not found! Creating a new one!", timetable.getBusName());
            bus = new Bus();
            bus.setBusRoutes(new ArrayList<>());
        }
        bus.setBusName(timetable.getBusName());
        return bus;
    }
}

File: src/main/kotlin/org/zenbot/szolnok/timetable/batch/step/bus/processor/CreateBusFromTimetableItemProcessorHelper.kt
package org.zenbot.szolnok.timetable.batch.step.bus.processor

import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component
import org.zenbot.szolnok.timetable.dao.BusRepository
import org.zenbot.szolnok.timetable.domain.Bus
import org.zenbot.szolnok.timetable.domain.Timetable

@Component
class CreateBusFromTimetableItemProcessorHelper(private val busRepository: BusRepository) {

    private val log = LoggerFactory.getLogger(CreateBusFromTimetableItemProcessorHelper::class.java)

    fun createBusFromTimetable(timetable: Timetable): Bus {
        log.debug("Fetching bus=[#{}] from database", timetable.busName)
        val bus = busRepository.findByBusName(timetable.busName)
        bus.busName = timetable.busName
        return bus
    }
}
